File: MovieCatalogQw/app/src/test/java/com/unlistedi/moviecatalogqw/movies/MoviesViewModelTest.java
package com.unlistedi.moviecatalogqw.movies;
import com.unlistedi.moviecatalogqw.data.MovieEntity;
import com.unlistedi.moviecatalogqw.ui.movies.MoviesViewModel;
import org.junit.Before;
import org.junit.Test;
import java.util.ArrayList;
import static org.junit.Assert.*;

public class MoviesViewModelTest {
    private MoviesViewModel moviesViewModel;

    @Before
    public void setUp(){
        moviesViewModel = new MoviesViewModel();
    }

    @Test
    public void getMovies(){
        ArrayList<MovieEntity> movies = moviesViewModel.getMovies();
        assertNotNull(movies);
        assertEquals(19, movies.size());
    }
}

File: MovieCatalogueKu/app/src/test/java/com/unlistedi/moviecatalogueku/ui/movies/MoviesViewModelTest.kt
package com.unlistedi.moviecatalogueku.ui.movies

import org.junit.Assert.*
import org.junit.Before
import org.junit.Test

class MoviesViewModelTest {
    private lateinit var moviesViewModel: MoviesViewModel

    @Before
    fun setUp(){
        moviesViewModel = MoviesViewModel()
    }

    @Test
    fun getMovies(){
        val movies = moviesViewModel.movies
        assertNotNull(movies)
        // Jumlah Asset yang disediakan hanya 19
        assertEquals(19, movies.size)
    }
}